frontend/packages/dev-console/src/components/import/section/ResourceSection.tsx

```diff
@@ -31,14 +41,88 @@ const createHelpText = (k8sModel: K8sKind, helpText: string) => {
 const ResourceSection: React.FC<ResourceSectionProps> = ({ flags }) => {
   const { t } = useTranslation();
   const [field] = useField<Resources[]>('resourceTypesNotValid');
+  const fieldName = 'resources';
+  const {
+    values: { formType },
+    setFieldValue,
+  } = useFormikContext<FormikValues>();
   const invalidTypes = field.value || [];
+
   const knativeServiceAccess = useAccessReview({
     group: ServiceModel.apiGroup,
     resource: ServiceModel.plural,
     namespace: getActiveNamespace(),
     verb: 'create',
   });
+  const canIncludeKnative =
+    !invalidTypes.includes(Resources.KnativeService) &&
+    flags[FLAG_KNATIVE_SERVING_SERVICE] &&
+    knativeServiceAccess;
+
+  const defaultResourceType = canIncludeKnative ? Resources.KnativeService : Resources.Kubernetes;
+
+  const [
+    preferredResourceType,
+    setPreferredResourceType,
+    preferredResourceTypeLoaded,
+  ] = useUserSettings<string>(PREFERRED_RESOURCE_TYPE_USER_SETTING_KEY, defaultResourceType, true);
+
+  const [resourceType, setResourceType, resourceTypeLoaded] = useUserSettingsCompatibility<string>(
+    LAST_RESOURCE_TYPE_STORAGE_KEY,
+    PREFERRED_RESOURCE_TYPE_LOCAL_STORAGE_KEY,
+    defaultResourceType,
+    true,
+  );
+
+  React.useEffect(() => {
+    if (
+      formType !== 'edit' &&
+      !flags[FLAG_KNATIVE_SERVING_SERVICE] &&
+      preferredResourceTypeLoaded &&
+      resourceTypeLoaded &&
+      (preferredResourceType === Resources.KnativeService ||
+        resourceType === Resources.KnativeService)
+    ) {
+      setResourceType(Resources.Kubernetes);
+      setPreferredResourceType(Resources.Kubernetes);
```

Comment:
@vikram-raj By the previous comment I meant we should have two cases here:

1. When preferredResourceType is latest and resourceType is KnativeService
2. When preferredResourceType KnativeService.

In 1st case only resourceType should be set to Kubernetes leaving the preferredResourceType as latest since this option is not removed from the list unlike Serverless which is removed when the operator is uninstalled. 

And in the 2nd  case we need to revert to default as the selected option is removed from the list. So preferredResourceType should be set to Kubernetes and it does not matter if we reset resourceType or not as it is not used unless the preferredResourceType is latest as per the code. 


@serenamarie125 @beaumorley what should happen when latest is selected but serverless operator is not installed anymore? Should latest now point to the default value which is Deployment or the selected option in the resource type field should change from latest to Deployment?

Reply:
@divyanshiGupta @vikram-raj 
Your recommendation to default back to Deployment sounds good to me. I'll double check with @serenamarie125 and we'll let you know today.

Reply:
@divyanshiGupta I updated the PR as per the @beaumorley input. PTAL. Thanks.